### PR TITLE
Fix/devtools object prototype

### LINKED ISSUE #47247: Devtools Console: `<prototype>` of an object expands to the object again
**Describe the bug:**
From #39858. See title.

**To Reproduce:**
Get an object output to the console and then try to expand it.
Typical result with `document`:
Image

**Platform:**
Win 10

### PR DESCRIPTION
**Pull request by sahilnyk**
When an object is printed in Servo DevTools it now sends its actual prototype to Firefox instead of sending the same object again. This stops the `<prototype>` entry from repeating forever.

Added a regression test to make sure the prototype does not repeat.

Testing: `./mach test-devtools test_console_log_object_prototype_does_not_repeat -N 0`

Fixes: #47247

Screenshot from 2026-09-19 14-42-06

### CODE REVIEW

File: components/script/dom/console.rs [hunk 1]
```diff
@@ -235,6 +237,47 @@ fn console_argument_from_handle_value(
...
+        ESClass::Map => "Map",
+        ESClass::Function => "Function",
+        ESClass::Other => return None,
+        _ => return None,
+    };
+
+    Some(Box::new(DebuggerValue::ObjectValue {
+        actor: None,
+        class: class.to_owned(),
+        own_property_length: None,
```
Comment (jdm) [inline components/script/dom/console.rs]: I'm not sure that hard coding these values is correct. We could have something like:
```js
const personPrototype = {
  greet() {
    console.log("hello!");
  },
};

const carl = Object.create(personPrototype);
```
and the prototype of carl would have an owned property. We should try to reuse the existing code that serializes objects into debugger values instead.

**Review by jdm — CHANGES_REQUESTED** [review]
Thanks for writing the test!

File: components/script/dom/console.rs (hunk 1 above)
Comment (sahilnyk) [reply components/script/dom/console.rs]: Addressed all the changes.

File: components/script/dom/console.rs [hunk 2]
```diff
@@ -204,7 +204,9 @@ fn console_argument_from_handle_value(
             }
 
             seen.push(handle_value.asBits_);
+            rooted!(&in(cx) let object = handle_value.to_object());
             let console_object = console_object_from_handle_value(cx, handle_value, seen);
+            let prototype = object_prototype_debugger_value(cx, object.handle(), seen);
```
Comment (eerii) [inline components/script/dom/console.rs]: We should return `Err` from here if we got `None` from this function, otherwise we are not handling the exception correctly. In this context `None` means that `JS_GetPrototype` failed (if it succeded but the prototype is null, it will be `Some(...)`).

**Review by eerii — CHANGES_REQUESTED** [review]
This looks better after the last changes, thanks! Left a small comment regarding exceptions.

One other thing is that this only applies to `console.log`. There's another way of displaying values in the debugger (using the value directly), and that uses the `createValueGrip` function. Not saying that it needs to be done in this PR, but a nice followup would be sending the `prototype` from it too, and making the other path work correctly.

```js
> console.log(a)
// good prototype
> a
// now always returns null
```

File: components/script/dom/console.rs (hunk 2 above)
Comment (sahilnyk) [reply components/script/dom/console.rs]: I handled that prototype lookup exception also added a test for it so if spidermonkey fails to get the prototype servo now shows error instead of null. for createValueGrip i havent include that yet.. if you allow then i can open a new pr for it (referencing this).

Comment (eerii) [reply components/script/dom/console.rs]: Yes, a new PR is better for this :)

**Review by eerii — APPROVED** [review]
Thank you!
